src/auth/jwt/mod.rs

```diff
@@ -0,0 +1,248 @@
+mod validation;
+
+use std::collections::HashSet;
+use std::fs;
+use std::num::NonZeroU64;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use headers::authorization::Bearer;
+use headers::{Authorization, HeaderMapExt};
+use jwtk::jwk::{JwkSet, JwkSetVerifier, RemoteJwksVerifier};
+use jwtk::HeaderAndClaims;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+use self::validation::{validate_aud, validate_iss};
+use super::base::{AuthError, AuthProvider};
+use crate::helpers::config_path::config_path;
+use crate::http::RequestContext;
+use crate::valid::{Valid, ValidationError};
+
+mod remote {
+  use std::num::NonZeroU64;
+
+  pub fn default_max_age() -> NonZeroU64 {
+    NonZeroU64::new(5 * 60 * 1000).unwrap()
+  }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub enum JwtProviderJwksOptions {
+  File(PathBuf),
+  #[serde(rename_all = "camelCase")]
+  Remote {
+    url: String,
+    #[serde(default = "remote::default_max_age")]
+    max_age: NonZeroU64,
+  },
+}
```

Comment:
```suggestion
  Constant {key: String} // to specify JWT key as a literal or an env variable.
}
```

This should support mustache templates.